File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) 
    {
        unordered_map<char,int> um;
        
        for(auto &it: s)um[it]++;
        
        for(auto &ch: t)
        {
            um[ch]--;
            if(um[ch]==0)um.erase(ch);
        }
        
        return um.empty();
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
from collections import Counter
class Solution(object):
    def isAnagram(self, s, t):
        
        if len(s)!=len(t):
            return False;
            
        mp1 = Counter(s)
        
        for ch in t:
            mp1[ch]-=1
            if mp1[ch]==0:
                del mp1[ch]
                
        
        return len(mp1)==0
        
        